### DISCUSSION TITLE
Jest test antd component render

### DISCUSSION THREAD

**Discussion by huypham1411** [ant-design/ant-design#46120] [Q&A] [answered]
Hi, I'm new to setting up Jest in NextJS so I wonder what I've missed in the configuration
here is my jest code to test if the button can render without any issue
```
import { render, screen } from '@testing-library/react';
import { Button } from 'antd';
import React from 'react';

jest.mock('next/navigation', () => ({
  useRouter: () => ({
    route: '/',
    pathname: '/',
    query: {},
    asPath: '/',
  }),
}));

describe('Login', () => {
  it('render page', () => {
    render(<Button />);
    const heading = 'render';
    expect(heading).toEqual('render');
  });
});

```

here is my jest.config.js
```
const nextJest = require('next/jest');

const createJestConfig = nextJest({
  // Provide the path to your Next.js app to load next.config.js and .env files in your test environment
  dir: './',
});

// Add any custom config to be passed to Jest
const customJestConfig = {
  setupFilesAfterEnv: ['<rootDir>/jest.setup.js'],
  testEnvironment: 'jest-environment-jsdom',
  preset: 'ts-jest',
  transformIgnorePatterns: [
    'node_modules/(?!@azure/msal-react)',
    'node_modules/(?!antd)/',
  ],
  transform: {
    '^.+\\.jsx?$': 'babel-jest',
    '^.+\\.(js|jsx|ts|tsx)$': 'ts-jest',
  },
  moduleNameMapper: {
    '@azure/msal-react': '<rootDir>/mocked/module.js',
  },
};

// createJestConfig is exported this way to ensure that next/jest can load the Next.js config which is async
module.exports = createJestConfig(customJestConfig);

```
when I run the code I got this error message:

>  \node_modules\antd\es\button\index.js:3
>     import Button from './button';
>     ^^^^^^
> 
>     SyntaxError: Cannot use import statement outside a module
> 
>       13 |   }),
>       14 | }));
>     > 15 |
>          | ^
>       16 | describe('Login', () => {
>       17 |   it('render page', () => {
>       18 |     render(<Button />);
> 
>       at Runtime.createScriptFromCode (node_modules/jest-runtime/build/index.js:1505:14)
>       at Object.<anonymous> (__tests__/index.test.tsx:15:56)
> 
> Test Suites: 1 failed, 1 total
> Tests:       0 total
> Snapshots:   0 total
> Time:        2.311 s

I've followed some guide based on the nextjs with jest example hope someone can help, appreciate

**Comment by huypham1411** [comment, marked as answer]
so I've solved my issue, here is the post that I posted with example and setup [https://github.com/vercel/next.js/discussions/58994](url) hope this help

**Reply by SushantG2710** [reply]
404 Error on page
